src/css/showHide.js

```diff
@@ -4,6 +4,23 @@ define( [
 	"../css/var/isHidden"
 ], function( jQuery, dataPriv, isHidden ) {
 
+var defaultDisplayMap = {};
+
+function setDefaultDisplay( elem ) {
+	var doc = elem.ownerDocument,
+		nodeName = elem.nodeName,
+		temp = doc.body.appendChild( doc.createElement( nodeName ) ),
+		display = jQuery.css( temp, "display" );
+
+	temp.parentNode.removeChild( temp );
+
+	if ( display === "none" ) {
+		display = "block";
```

Comment:
As you said above @markelog it misses a couple of cases, but if those aren't super common this might just work well enough. If the person writing a rule like `li { display: none }` is the author of the page they can just rearrange the HTML and CSS a bit to fix that. I think the case that would kill things would be plugins that need to force-hide content. There, hiding by tag name isn't as likely?

![bold-strategy](https://cloud.githubusercontent.com/assets/157858/12303575/1ca41904-b9f9-11e5-8ab9-6c2aebfc5392.jpg)